#goal
---
#result
bool push_success
---
#feedback
string ongoing_state